crates/uv-workspace/src/workspace.rs

```diff
@@ -728,6 +728,13 @@ impl Workspace {
                 let member_root = std::path::absolute(&member_root)
                     .map_err(WorkspaceError::Normalize)?
                     .clone();
+                if !fs_err::metadata(&member_root)?.is_dir() {
+                    warn!(
+                        "Ignoring non-directory workspace member: `{}`",
+                        member_root.simplified_display()
+                    );
+                    continue;
```

Comment:
Could we do this inside of the `if err.kind() == std::io::ErrorKind::NotFound` branch instead, to avoid paying this cost in the common case?